particles/Tutorial/Kotlin/Blackjack/DealingShoe.kt

```diff
@@ -0,0 +1,59 @@
+package arcs.tutorials.blackjack
+
+import kotlin.random.Random
+import arcs.sdk.Handle
+
+import arcs.sdk.Utils.log
+
+class DealingShoe : AbstractDealingShoe() {
+    val cardPresent = "x"
+    val cardAbsent = "-"
+    val numDecks = 1
+    val totalCards = numDecks * 52
+    val emptyDeck = cardAbsent.repeat(totalCards)
+
+    override fun getTemplate(slotName: String) = """
+        Card is <span>{{nextCard}}</span>
+     """.trimIndent()
+
+    override fun populateModel(slotName: String, model: Map<String, Any>): Map<String, Any> {
+        return model + mapOf("nextCard" to nextCard.toString())
+    }
+
+    override fun onHandleUpdate(handle: Handle) {
+        if (handle.name != "cardRequest") { return }
+        val request = cardRequest.get()
+        if (request == null) { return }
```

Comment:
I think an elvis operator could be used here as well.

Reply:
Done!